test/tower_honeybadger_test.exs

```diff
@@ -159,6 +159,53 @@ defmodule TowerHoneybadgerTest do
     end)
   end
 
+  test "reports throw with Bandit", %{bypass: bypass} do
+    # An ephemeral port hopefully not being in the host running this code
+    plug_port = 51111
+    url = "http://127.0.0.1:#{plug_port}/uncaught-throw"
+
+    waiting_for(fn done ->
+      Bypass.expect_once(bypass, "POST", "/notices", fn conn ->
+        {:ok, body, conn} = Plug.Conn.read_body(conn)
+
+        assert(
+          {
+            :ok,
+            %{
+              "error" => %{
+                # An exit instead of a throw because Bandit doesn't handle throw's
+                # for the moment. See: https://github.com/mtrudel/bandit/pull/396.
+                "class" => "(exit)",
+                "message" => "bad return value: \"from inside a plug\"",
```

Comment:
After digging a bit deeper on why a `throw` within Bandit ends up being a "bad return value" exit, realized that it's because, given Bandit doesn't handle throws, and that the plug call runs inside a `:gen_server`, the thrown value gets interpreted as the `:gen_server` callback return value. See https://github.com/mtrudel/bandit/issues/410.

So given that we're testing with throwing a string here, and that is an invalid `:gen_server` return value, we do get termination of the gen_server process and an exit event.

However, if the thrown value were to coincide with a valid `:gen_server` return value, no event would be raised. Extremely unlikely I guess, but technically possible.